lib/src/layer/tile_layer/tile_provider/base_tile_provider.dart

```diff
@@ -17,23 +17,24 @@ abstract class TileProvider {
   });
 
   /// Retrieve a tile as an image, based on it's coordinates and the current [TileLayerOptions]
-  ImageProvider getImage(Coords coords, TileLayer options);
+  ImageProvider getImage(TileCoordinate coords, TileLayer options);
 
   /// Called when the [TileLayerWidget] is disposed
   void dispose() {}
 
-  String _getTileUrl(String urlTemplate, Coords coords, TileLayer options) {
+  String _getTileUrl(
+      String urlTemplate, TileCoordinate coords, TileLayer options) {
     final z = _getZoomForUrl(coords, options);
 
     final data = <String, String>{
-      'x': coords.x.round().toString(),
-      'y': coords.y.round().toString(),
-      'z': z.round().toString(),
+      'x': coords.x.toString(),
+      'y': coords.y.toString(),
+      'z': z.toString(),
```

Comment:
No rounding needed as they are now ints. The existence of rounding shows that tile coordinates can/should be ints instead of doubles as previously.